meteor/client/ui/FloatingInspectors/SplitsFloatingInspector.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useMemo, memo } from 'react'
+import ClassNames from 'classnames'
+
+import { FloatingInspector } from '../FloatingInspector'
+import { RundownUtils } from '../../lib/rundown'
+import { SplitRole, SplitSubItem, SplitsSourceRenderer } from '../SegmentTimeline/Renderers/SplitsSourceRenderer'
+import { SplitsContent } from '@sofie-automation/blueprints-integration'
+
+interface IProps {
+	typeClass?: string
+	showMiniInspector: boolean
+	itemElement: HTMLDivElement | null
+	floatingInspectorStyle: React.CSSProperties
+	content: SplitsContent
+}
+
+const RenderSplitPreview = memo(function RenderSplitPreview({ subItems }: { subItems: SplitSubItem[] }) {
+	return (
+		<div className="video-preview">
+			{subItems.reverse().map((item, index, array) => {
+				return (
+					<div
+						className={ClassNames(
+							'video-preview',
+							RundownUtils.getSourceLayerClassName(item.type),
+							{
+								background: item.role === SplitRole.ART,
+								box: item.role === SplitRole.BOX,
+							},
+							{
+								second: array.length > 1 && index > 0 && item.type === array[index - 1].type,
+							}
+						)}
+						key={item._id + '-preview'}
+						style={{
+							left: ((item.content && item.content.x) * 100).toString() + '%',
```

Comment:
What happens with these properties if item.content or item.content.x is undefined?

Reply:
It's never going to be undefined, since if no geometry is provided, a default two-box setup will be used - but I agree that since typings allow for it to be undefined, it should be handled! And yeah, I suppose that's going to end up being "NaN%" - so it's very good feedback. Thanks again for reminding me about nullish coalescing operator! It allows for setting this up with a reasonable fallback. I was able to tighten the typings for `SplitSubItem.content` as well, so thanks for making me take a second look at this! 👍